clientd3d/draw3d.c

```diff
@@ -327,34 +333,33 @@ void DrawPostOverlayEffects(room_type* room, Draw3DParams* params)
 /************************************************************************/
 void DrawViewTreatment()
 {
-	//	Draw view elements (edge treatment).
-	//	added by ajw.
-	int i;
-	int iOffset = 0;
+   if (gD3DDriverProfile.bSoftwareRenderer == TRUE)
+   {
+      //	Draw view elements (edge treatment): added by ajw.
+      int i;
+      int iOffset = 0;
 
-//	HDC hdcTarget = ( stretchfactor == 1 ) ? gBitsDC : gBufferDC;
-	BYTE* pBitsTarget = ( stretchfactor == 1 ) ? gBits : gBufferBits;
-	int iWidthTarget = ( stretchfactor == 1 ) ? MAXX : MAXX*2;
+      BYTE* pBitsTarget = (stretchfactor == 1) ? gBits : gBufferBits;
+      int iWidthTarget = (stretchfactor == 1) ? MAXX : MAXX * 2;
 
-//	if (state == STATE_GAME && (GameGetState() == GAME_PLAY || GameGetState() == GAME_SELECT))
-	if (GetFocus() == hMain)
-		iOffset = 4;
+      if (GetFocus() == hMain)
+         iOffset = 4;
 
-	for( i = iOffset; i < iOffset + ( NUM_VIEW_ELEMENTS / 2 ); i++ )
-	{
-		BitCopy( pBitsTarget, iWidthTarget, ViewElements[i].x, ViewElements[i].y, ViewElements[i].width, ViewElements[i].height,
-					ViewElements[i].bits, 0, 0, DIBWIDTH( ViewElements[i].width ), OBB_FLIP | OBB_TRANSPARENT );
-//		OffscreenBitBlt( hdcTarget, ViewElements[i].x, ViewElements[i].y, ViewElements[i].width, ViewElements[i].height,
-//							ViewElements[i].bits, 0, 0, DIBWIDTH( ViewElements[i].width ), OBB_COPY | OBB_FLIP | OBB_TRANSPARENT );
-		GdiFlush();
-	}
-	//	Ensure that border, which covers up parts of view treatment, is drawn.
-	DrawGridBorder();
-	GdiFlush();
-	GdiFlush();
-	GdiFlush();
-	GdiFlush();
-	GdiFlush();
+      for (i = iOffset; i < iOffset + (NUM_VIEW_ELEMENTS / 2); i++)
+      {
+         BitCopy(pBitsTarget, iWidthTarget, ViewElements[i].x, ViewElements[i].y, ViewElements[i].width, ViewElements[i].height,
+            ViewElements[i].bits, 0, 0, DIBWIDTH(ViewElements[i].width), OBB_FLIP | OBB_TRANSPARENT);
+         GdiFlush();
+      }
+   }
+
+   //	Ensure that border, which covers up parts of view treatment, is drawn.
+   DrawGridBorder();
+   GdiFlush();
```

Comment:
Any idea why there are 5 GdiFlush calls here?  I know they were already here, but it's weird.

Reply:
I have no knowledge on why there are 5 calls here but I've just removed these and see no obvious issues in doing so